### PR TITLE
nose2pytest: init at 1.0.12

### PR DESCRIPTION
**Pull request by emilazy**
## Description of changes

They said nobody could get a new package using nose into Nixpkgs… they were wrong!!!

Obviously there is an element of farce here, but I just used this to accomplish https://github.com/yandex/gixy/pull/146 and I suspect it may come in handy for others working on https://github.com/NixOS/nixpkgs/issues/326513 or who just have their own nose test suites they’d like to migrate. The upstream package declares that it doesn’t work on Python 3.12, I guess because the nose‐using tests won’t natively run on that version, but it seems to work fine for me.

Of course, I’ll remove the nose `nativeCheckInput` and just disable the tests when the time comes. (Or now, if other people don’t find it as amusing as I do.)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1933,6 +1933,8 @@ with pkgs;
     postgresql = postgresql_14;
   };
 
+  nose2pytest = with python3Packages; toPythonApplication nose2pytest;
```
Comment (drupol) [inline pkgs/top-level/all-packages.nix:1936]: Shouldn't we use `pkgs/by-name` for this?

Comment (emilazy) [reply pkgs/top-level/all-packages.nix:1936]: It’s also a library; its output can depend on itself for complex cases. (Which is probably not that useful, but hey.)

Or are you saying that I should put a stub in `pkgs/by-name` that just calls `toPythonApplication` on the library? Does anything else do that?

Comment (drupol) [reply pkgs/top-level/all-packages.nix:1936]: > Or are you saying that I should put a stub in pkgs/by-name that just calls toPythonApplication on the library?

Yes precisely.

**Review by drupol — APPROVED** [review]
Diff LGTM !

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (emilazy) [reply pkgs/top-level/all-packages.nix]: Well… sure, why not. Pushed.

Comment (natsukium) [reply pkgs/top-level/all-packages.nix]: In most cases, we simply add a line to toplevel/all-packages.nix, but that is ok.

**Review by natsukium — APPROVED** [review]
I don't know how many active upstreams actually use nose in 2024, but it's fine for me to package it as long as the other packages in nixpkgs don't depend on it.

**Pull request merged**